frontend/packages/sdk/src/contracts/PinkBlueprint.ts

```diff
@@ -113,9 +113,22 @@ export class PinkBlueprintSubmittableResult extends SubmittableResult {
       if (!contractId) {
         throw new Error('Failed to find contract ID in events, maybe instantiate failed.')
       }
+      const logger = this.registry.loggerContract
 
       const t0 = new Date().getTime()
       while (true) {
+        if (logger) {
+          const { records } = await logger.tail(10, { contract: contractId })
+          if (
+            records.length > 0 &&
+            records[0].type === 'Log' &&
+            records[0].execMode === 'transaction' &&
+            records[0].message !== 'instantiated'
+          ) {
+            throw new Error(records[0].message)
```

Comment:
If the contract print a log in the constructor, would it be treated as error here?

Reply:
Good point. Let me change it to matched keyword `instantiate failed`